tools/accuracy_checker/accuracy_checker/data_readers/data_reader.py

```diff
@@ -180,8 +179,163 @@ def read(self, data_id):
+
+        if mode is None:
+            mode = 'RGB' if numch == 3 else 'RGBA'
+
+        if mode not in ['RGB', 'RGBA', 'YCbCr', 'CMYK']:
+            raise ValueError("Mode is unknown or incompatible with input array shape.")
+
+        if mode in ['RGB', 'YCbCr'] and numch != 3:
+            raise ValueError("Invalid array shape for mode.")
+        if mode in ['RGBA', 'CMYK'] and numch != 4:
+            raise ValueError("Invalid array shape for mode.")
+
+        # Here we know data and mode is correct
+        image = Image.frombytes(mode, shape, strdata)
+        return image
+
+    @staticmethod
+    def _to_image(arr, high=255, low=0, cmin=None, cmax=None, pal=None, mode=None, channel_axis=None):
+        data = np.asarray(arr)
+        if np.iscomplexobj(data):
+            raise ValueError("Cannot convert a complex-valued array.")
+        shape = list(data.shape)
+        valid = len(shape) == 2 or ((len(shape) == 3) and ((3 in shape) or (4 in shape)))
+        if not valid:
+            raise ValueError("'arr' does not have a suitable array shape for any mode.")
+        if len(shape) == 2:
+            return ScipyImageReader._process_2d(data, shape, mode, pal, high, low, cmax, cmin)
+        return ScipyImageReader._process_3d(data, shape, mode, channel_axis, high, low, cmin, cmax)
+
+    @staticmethod
+    def _imread(name):
+        # reimplementation scipy.misc.imread
+        image = Image.open(name)
+
+        return ScipyImageReader._from_image(image)
+
+    @staticmethod
+    def _bytescale(data, cmin=None, cmax=None, high=255, low=0):
+        if data.dtype == np.uint8:
+            return data
+
+        if high > 255:
+            raise ValueError("`high` should be less than or equal to 255.")
+        if low < 0:
+            raise ValueError("`low` should be greater than or equal to 0.")
+        if high < low:
+            raise ValueError("`high` should be greater than or equal to `low`.")
+        cmin = cmin or data.min()
+        cmax = cmax or data.max()
+
+        cscale = cmax - cmin
+        if cscale < 0:
+            raise ValueError("`cmax` should be larger than `cmin`.")
+        if cscale == 0:
+            cscale = 1
+
+        scale = float(high - low) / cscale
+        bytedata = (data - cmin) * scale + low
+
+        return (bytedata.clip(low, high) + 0.5).astype(np.uint8)
+
+    @staticmethod
+    def imresize(arr, size, interp='bilinear', mode=None):
+        im = ScipyImageReader._to_image(arr, mode=mode)
+        ts = type(size)
+        if np.issubdtype(ts, np.signedinteger):
+            percent = size / 100.0
+            size = tuple((np.array(im.size) * percent).astype(int))
+        elif np.issubdtype(type(size), np.floating):
+            size = tuple((np.array(im.size) * size).astype(int))
+        else:
+            size = (size[1], size[0])
+        func = {'nearest': 0, 'lanczos': 1, 'bilinear': 2, 'bicubic': 3, 'cubic': 3}
+        imnew = im.resize(size, resample=func[interp])
```

Comment:
I think this can be simplified a lot, given that `size` is always a tuple, `interp` is always `'nearest'` and `mode` is always `None`.